praetorian_cli/sdk/entities/webpage.py

```
@@ -0,0 +1,144 @@
+            >>> pages, offset = sdk.webpage.list()
+
+            >>> # List pages for specific WebApplication
+            >>> pages, offset = sdk.webpage.list(
+            ...     webapp_key="#asset#webapp#https://app.example.com#https://app.example.com")
+
+        **WebPage Filtering:**
+            - parent_key: Filters by parent WebApplication
+            - filter: Filters by specific URL
+        """
+        if parent_key and not parent_key.startswith('#webapplication#'):
+            raise Exception("Invalid WebApplication key format")
```

Ruff (0.12.2)

Create your own exception

(TRY002)